zboxcore/sdk/sdk.go

```diff
@@ -1475,38 +1475,45 @@ func GetAllocationMinLock(
 }
 
 // calculateAllocationFileOptions calculates the FileOptions 16-bit mask given the user input
-func calculateAllocationFileOptions(initial uint16, fop *FileOptionsParameters) uint16 {
+func calculateAllocationFileOptions(initial uint16, fop *FileOptionsParameters) (bool, uint16) {
 	if fop == nil {
-		return initial
+		return false, initial
 	}
 
 	mask := initial
+	changed := false
 
 	if fop.ForbidUpload.Changed {
+		changed = true
 		mask = updateMaskBit(mask, 0, !fop.ForbidUpload.Value)
 	}
 
 	if fop.ForbidDelete.Changed {
+		changed = true
 		mask = updateMaskBit(mask, 1, !fop.ForbidDelete.Value)
 	}
 
 	if fop.ForbidUpdate.Changed {
+		changed = true
 		mask = updateMaskBit(mask, 2, !fop.ForbidUpdate.Value)
 	}
 
 	if fop.ForbidMove.Changed {
+		changed = true
 		mask = updateMaskBit(mask, 3, !fop.ForbidMove.Value)
 	}
 
 	if fop.ForbidCopy.Changed {
+		changed = true
 		mask = updateMaskBit(mask, 4, !fop.ForbidCopy.Value)
 	}
 
 	if fop.ForbidRename.Changed {
+		changed = true
 		mask = updateMaskBit(mask, 5, !fop.ForbidRename.Value)
 	}
 
-	return mask
+	return changed, mask
```

Comment:
is changed the same as  `initial == mask`?

Reply:
I got your point, I'll use `initial == mask` instead, nice one btw 👌